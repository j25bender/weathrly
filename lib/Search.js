import React from 'react';
import App from '../lib/App.js';
import '../styles/Search.css';
import autoData from '../lib/AutoCompleteData.js';
import { Trie } from '@j25bender/complete-me';

export default class Search extends React.Component {
    constructor () {
        super();
        this.trie = new Trie();
        this.trie.populate(autoData.data)

        this.state = {
            inputValue: ''
        }
        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleInputChange (e) {
        this.setState({
            inputValue: e.target.value
        })        
    }

    handleBtnClick () {
        if(isNaN(this.state.inputValue) 
            && autoData.data.includes(this.state.inputValue) 
            || this.state.inputValue.length === 5 && !isNaN(this.state.inputValue)) {

            this.props.searchValue(this.state.inputValue);
            this.setState({ inputValue: this.state.inputValue });
            localStorage.setItem('location', this.state.inputValue);
        } else { 
            alert('Invalid Input:\n\nPlease Enter City, State OR Zip Code\n\ne.g. Denver, CO OR 80202');
            this.setState({ inputValue: '' });
            document.getElementById('searchInput').focus();
        }       
    }

    render() {
        return (
            <div className="searchContainer">
                <div className="welcome-transition">
                    <div className="container">
                    <h4 className="welcomeText">Welcome to Weathrly!</h4>
                    <input id="searchInput" 
                           value={this.state.inputValue}
                           type="text"
                           onChange={this.handleInputChange}
                           autoFocus
                    />
                    <button onClick={this.handleBtnClick}>Search by City or Zip</button>
                    </div>
                </div>
            </div>
        )
    }
}