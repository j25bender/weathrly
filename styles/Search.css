.searchContainer {
    height: 100vh;
}

.welcome-transition {
    text-align: center;
    border: 0.02rem solid ghostwhite;    
    top: 60px;
    width: 90vw;
    height: 60vh;
    margin: auto;
}

.container {
    position: relative;
    top: 50%;
    transform: translateY(-50%);
}