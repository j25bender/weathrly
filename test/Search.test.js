import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from "../lib/Search";
import cleanData from '../lib/cleanData';
import mockData from '../lib/mockData';
import 'jest-localstorage-mock';

describe('Search Tests ', () => {
    let filteredData;
    let wrapper;
    
    beforeEach(() => {
        filteredData = cleanData(mockData);
        wrapper = mount(<Search />);
    });

    it('Search component should exist', () => {
        expect(wrapper).toBeDefined()
        expect(wrapper.find('.welcomeText').text()).toEqual('Welcome to Weathrly!')        
    });

    it('Search should render 1 Search component', () => {
        expect(wrapper.find('Search').length).toEqual(1);
        expect(Search).toMatchSnapshot();
    });
  
    it('Should call handleBtnClick when button is clicked', () => {
        const submitButton = wrapper.find('button')
        const inputValue = wrapper.find('input')

        wrapper.instance().handleBtnClick = jest.fn()
        
        inputValue.simulate('change', { target: { value: 'Denver, CO'}})
        submitButton.simulate('click')
        
        expect(wrapper.instance().handleBtnClick).toHaveBeenCalledTimes(1)

        inputValue.simulate('change', { target: { value: 'Boston, MA'}})
        submitButton.simulate('click')

        expect(wrapper.instance().handleBtnClick).toHaveBeenCalledTimes(2)
    });

    // error handling add className div to error find it
});