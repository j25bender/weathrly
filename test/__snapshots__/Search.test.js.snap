// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Tests  Search should render 1 Search component 1`] = `[Function]`;
